modules/contact/src/constraints/TangentialMortarLMMechanicalContact.C

```diff
@@ -88,7 +88,12 @@ TangentialMortarLMMechanicalContact<compute_stage>::computeQpResidual(Moose::Mor
           if (tangential_velocity * _lambda[_qp] < 0)
             a = -std::abs(tangential_velocity);
           else
+          {
+            if (tangential_velocity == 0)
```

Comment:
Should this be a strict equality check, or use a fuzzy comparison?

Reply:
It should be strict. If it's not nonzero at all, then the absolute value derivative is well defined

Reply:
OK.  I think this PR looks pretty good. Just get the tests passing...

Make sure you're referencing an appropriate issue for adding the scaling options.